### PR TITLE
Remove userData from bufferWrapOrUnwrap struct

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Description

Remove UserData from BufferWrapOrUnwrapParams, since there's no customization available in the ERC4626 buffers.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/pool-weighted/foundry.toml [hunk 1]
```diff
@@ -29,11 +29,11 @@ evm_version = 'cancun'
 ignored_error_codes = [2394, 5574, 3860] # Transient storage, code size
 
 [fuzz]
-runs = 10000
+runs = 100
```
Comment (EndymionJkb) [inline pkg/pool-weighted/foundry.toml:32]: Is this related, or left over from something else?

**Review by EndymionJkb — APPROVED** [review]
Great savings (and we need it); 1024 pushes it over.

File: pkg/pool-weighted/foundry.toml (hunk 1 above)
Comment (jubeira) [reply pkg/pool-weighted/foundry.toml:32]: I'd revert this one, seems unrelated.

**Review by jubeira — APPROVED** [review]
Nice catch! Good simplification, let's :ship:
